__tests__/app.test.js

```diff
@@ -12,15 +12,96 @@ const app = require("../index");
 beforeEach(() => seed({ topicData, articleData, userData, commentData }));
 
 afterAll(() => db.end());
+describe("Error tests", () => {
+  describe("testing 404 endpoint when given non existing URL", () => {
+    test("should output 404 and a message when givien a invalid url", () => {
+      return request(app)
+        .get("/api/badURL")
+        .expect(404)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe("Invalid Path !");
+        });
+    });
+  });
+  describe("test for non existing id on app GET /api/articles/:article_id", () => {
+    const nonExistingId = 99999999;
+    test("should output a 404 and a message when given non=existent id", () => {
+      return request(app)
+        .get(`/api/articles/${nonExistingId}`)
+        .expect(404)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe("This article id does not exist !");
+        });
+    });
+  });
 
-describe("testing 404 endpoint", () => {
-  test("should output 404 and a message when givien a invalid url", () => {
-    return request(app)
-      .get("/api/badURL")
-      .expect(404)
-      .then(({ body: { msg } }) => {
-        expect(msg).toBe("Invalid Path !");
-      });
+  describe("test for get /api/articles/:article_id/comments is given a non existing id", () => {
+    const nonExistingId = 99999999;
+    test("should output a 404 and a message when given non=existent id for comments endpoint", () => {
+      return request(app)
+        .get(`/api/articles/${nonExistingId}/comments`)
+        .expect(404)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe("This article id does not exist !");
+        });
+    });
+  });
+
+  describe("tests for incorrect input types", () => {
+    const badId = "thisisabadId";
+    test("should output 400 and a message stating the id is invalid", () => {
+      return request(app)
+        .get(`/api/articles/${badId}`)
+        .expect(400)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe("Your body or request has the wrong input type");
+        });
+    });
+    test("should output 400 and a message stating the id is invalid for comments endpoint", () => {
```

Comment:
Your test file would be much easier to navigate if the errors and happy paths where contained in a describe block pertaining to that particular endpoint :)

Reply:
I'll make that change on the current ticket i'm working on